reactioncommerce/reaction
# Fixed indicating products that are invalid for publication

Partly resolves #2690 ,

This PR adds a red badge at the top of products that have failed validation before being published. I added a `checkValidation` function in `gridItemControlContainer.js`  that uses _Simple Schema_ validation method to check validity of a product including it's variants before publishing it. And also added a condition to also add the appropriate badge for when a product is invalid in `gridItemControl.js` component.

#### **HOW TO TEST**
Some context before proceeding to test, so initially it was quite confusing for a user to differentiate between products that require to be published and those that required all fields to be completed as it indicated a green info badge on the affected products in the product grid. To understand what is trying to be fixed here, the test below is going to include how to create these 2 instances of products.

**How to test for the green badge**
```
- Login as Admin
- Create a product fill out all the fields required
- Navigate to the product grid
- Observe that the product you just created has a green badge at the top right corner.
```
**How to test for the red badge**
```
- Login as Admin
- Create a product but leave "Label" blank
- Navigate to the product grid
- Observe that the product you just created has a red badge at the top right corner.
```
